### ISSUE TITLE
Password for PEM file

### ISSUE THREAD

**Issue by gcartlidge** [dddomodossola/remi#468]
Hi All,

I'm currently designing a tool for another software to do background tasks that allows stopping the services for that software.

When starting the web server part of remi, I have a .pem file (that is password protected), a .key file and a .cer file (not password protected).

It works perfectly with the .cer and .key files.

I belive the certfile section of the server supports .pem files without password (as it just passes this to httpserver). But how would I define the PEM password while starting the remi instance?

The only way I can currently see of doing it at the moment is opening the file via PEM/SSL, plugging the password in, then generating a temp file of the non-password protected PEM file. Then loading this as certfile. 

Basically, is there an easier way for defing the password as a variable?

**Comment by gcartlidge** [comment]
Hi dddomodossola,

Im actually looking at the code now (I got bored).
It looks like there was a new argument introduced in SSL itself of 'password' when it does load_cert_chain.

It looks like it would be just adding this as an argument in the __init__ 's and start (of server), which would then pass it along to load_cert_chain: 
`SSLContext.load_cert_chain(certfile, keyfile=None, password=None) `
before doing a wrap_socket.

At the moment, i belive it works, but it prompts for the password in the terminal view while loading up.

Maybe for the future, but ill just use the above process with the temp files at the moment :)
